budgeteer-web-interface/src/main/java/org/wickedsource/budgeteer/web/pages/budgets/overview/BudgetsOverviewPage.java

```diff
@@ -41,17 +43,18 @@
     public BudgetsOverviewPage() {
         BudgetTagsModel tagsModel = new BudgetTagsModel(BudgeteerSession.get().getProjectId());
         if (BudgeteerSession.get().getBudgetFilter() == null) {
-            BudgetTagFilter filter = new BudgetTagFilter(new ArrayList<String>(), BudgeteerSession.get().getProjectId());
+            BudgetTagFilter filter = new BudgetTagFilter(new ArrayList<>(), BudgeteerSession.get().getProjectId());
             BudgeteerSession.get().setBudgetFilter(filter);
         }
+        add(new BudgetRemainingFilterPanel("remainingFilter", new RemainingBudgetFilterModel(BudgeteerSession.get().getProjectId())));
         add(new BudgetTagFilterPanel("tagFilter", tagsModel));
-        add(new BudgetOverviewTable("budgetTable", new FilteredBudgetModel(BudgeteerSession.get().getProjectId(), model(from(BudgeteerSession.get().getBudgetFilter()))), getBreadcrumbsModel()));
-
+        FilteredBudgetModel filteredBudgetModel = new FilteredBudgetModel(BudgeteerSession.get().getProjectId(), model(from(BudgeteerSession.get().getBudgetFilter())));
+        filteredBudgetModel.setRemainingFilterModel(model(from(BudgeteerSession.get().getBudgetRemainingFilter())));
```

Comment:
I find this confusing: is the model we get from `model(from(BudgeteerSession.get().getBudgetRemainingFilter()))` the same one we created earlier on line 49 for the `BudgetRemainingFilterPanel`?

Reply:
No, this is just a model we create from a Long which is saved in the BudgeteerSession.
Initially it is zero. The remaining filter model is just a wrapper around the value. (We need a LoadableDetachableModel for the FilterPanel)